keras_cv/models/object_detection_3d/center_pillar.py

```diff
@@ -11,127 +11,22 @@
-                    filters=self._per_class_prediction_size[i],
-                    kernel_size=(1, 1),
-                    name=f"head_{i + 1}",
-                )
-        else:
-            shared_layer = keras.layers.Conv2D(
-                filters=self._per_class_prediction_size[0],
-                kernel_size=(1, 1),
-                name="shared_head",
-            )
-            for i in range(num_classes):
-                self._heads[self._head_names[i]] = shared_layer
-
-    def call(self, feature: tf.Tensor, training: bool) -> List[tf.Tensor]:
-        del training
-        outputs = {}
-        for head_name in self._head_names:
-            outputs[head_name] = self._heads[head_name](feature)
-        return outputs
-
-
-class MultiClassHeatmapDecoder(keras.layers.Layer):
-    def __init__(
-        self,
-        num_classes,
-        num_head_bin: Sequence[int],
-        anchor_size: Sequence[Sequence[float]],
-        max_pool_size: Sequence[int],
-        max_num_box: Sequence[int],
-        heatmap_threshold: Sequence[float],
-        voxel_size: Sequence[float],
-        spatial_size: Sequence[float],
-        **kwargs,
-    ):
-        super().__init__(**kwargs)
-        self.num_classes = num_classes
-        self.class_ids = list(range(1, num_classes + 1))
-        self.num_head_bin = num_head_bin
-        self.anchor_size = anchor_size
-        self.max_pool_size = max_pool_size
-        self.max_num_box = max_num_box
-        self.heatmap_threshold = heatmap_threshold
-        self.voxel_size = voxel_size
-        self.spatial_size = spatial_size
-        self.decoders = {}
-        for i, class_id in enumerate(self.class_ids):
-            self.decoders[f"class_{class_id}"] = HeatmapDecoder(
-                class_id=class_id,
-                num_head_bin=self.num_head_bin[i],
-                anchor_size=self.anchor_size[i],
-                max_pool_size=self.max_pool_size[i],
-                max_num_box=self.max_num_box[i],
-                heatmap_threshold=self.heatmap_threshold[i],
-                voxel_size=self.voxel_size,
-                spatial_size=self.spatial_size,
-            )
-
-    def call(self, predictions):
-        box_predictions = []
-        class_predictions = []
-        box_confidence = []
-        for k, v in predictions.items():
-            boxes, classes, confidence = self.decoders[k](v)
-            box_predictions.append(boxes)
-            class_predictions.append(classes)
-            box_confidence.append(confidence)
-
-        return {
-            "3d_boxes": {
-                "boxes": tf.concat(box_predictions, axis=1),
-                "classes": tf.concat(class_predictions, axis=1),
-                "confidence": tf.concat(box_confidence, axis=1),
-            }
-        }
-
-
-class MultiHeadCenterPillar(keras.Model):
+class MultiHeadCenterPillar(Task):
     """Multi headed model based on CenterNet heatmap and PointPillar.
```

Comment:
Will it be clear to users what a voxel net is and how to make one?

Reply:
Probably not with the previous docstring, so I've edited it to point to our voxelization layer